src/server/game/Entities/Object/Object.h

```diff
@@ -396,7 +397,10 @@ class TC_GAME_API WorldObject : public Object, public WorldLocation
         void SummonCreatureGroup(uint8 group, std::list<TempSummon*>* list = nullptr);
 
         Creature*   FindNearestCreature(uint32 entry, float range, bool alive = true) const;
+        std::list<Creature*> FindAllCreaturesByEntry(uint32 entry, float range, bool alive = true) const;
```

Comment:
Is there any way making this return some sort of generic container like IEnumerable in C#?

Reply:
You retrieve a pointer to the boss within IsSummonedBy and call BossAI::JustSummoned, see the second link i pasted. Almost everything on Halion is not summoned by Halion

Reply:
Its not that simple, its not the problem of `IsSummonedBy` it is the problem of gameobject not being able to be "owner" of the summoned unit it has from the spell it (the gameobject) casts, which clearly the core does not support and as I see it requires a lot of work.